### PR TITLE
DATAUP-69 Create pull_request_template.md

### PR DESCRIPTION
**Pull request by bio-boris**
Adds a pull request template for future PRs as per https://kbase-jira.atlassian.net/browse/DATAUP-69

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Description of PR purpose/changes
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
```
Comment (ialarmedalien) [inline .github/pull_request_template.md:3]: Can we also have a link to the Jira ticket? It is so much easier just to click on a link than to have to find the ticket in Jira manually.

Comment (bio-boris) [reply .github/pull_request_template.md:3]: @ialarmedalien I added that at the very end. Where would you like this added?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- [ ] This change requires a documentation update
+
+# How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+- [ ] Instructions for how to test the PR
+
+# Dev Checklist:
+
```
Comment (eamahanna) [inline .github/pull_request_template.md:23]: For the dev checklist, do we want to include all of this or just what we agreed on as a team? 
https://sites.google.com/d/1hHn4AjWF19JdEjxkyMmrkBYIcMFZWxpp/p/1_xSUWHepJ3fRf64k4VC-Ex-2eKNPTebe/edit?pli=1

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+Optional Fixes # (issue)
+
+## Type of change
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+# How Has This Been Tested?
```
Comment (eamahanna) [inline .github/pull_request_template.md:16]: I think this section could maybe be `How to Test this PR` and have it include instructions for how reviewers can test the PR. This is just a nit pick though. You definitely can leave it which ever way you prefer.

File: .github/pull_request_template.md (hunk 1 above)
Comment (eamahanna) [reply .github/pull_request_template.md]: You could call it out in a section of it's own i.e. ## JIRA Story/Task, and have it in its own section so it is easy to find.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Description of PR purpose/changes
+
+* Please include a summary of the change and which issue is fixed. 
+* Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+## Jira Ticket / Issue #
+e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X
+
+## Type of change
+
```
Comment (eamahanna) [inline .github/pull_request_template.md:10]: Is the type of change section needed if we are linking to the jira ticket?

File: .github/pull_request_template.md (hunk 2 above)
Comment (bio-boris) [reply .github/pull_request_template.md:16]: I added that as a link

File: .github/pull_request_template.md (hunk 3 above)
Comment (bio-boris) [reply .github/pull_request_template.md]: Changed it a bit

File: .github/pull_request_template.md (hunk 4 above)
Comment (bio-boris) [reply .github/pull_request_template.md]: removed

File: .github/pull_request_template.md (hunk 1 above)
Comment (bio-boris) [reply .github/pull_request_template.md]: @eamahanna Done

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] Any dependent changes have been merged and published in downstream modules
+- [ ] Title of the PR contains Jira ticket followed by the description of the PR e.g. (DATAUP-69 Adds a PR template)
```
Comment (eamahanna) [inline .github/pull_request_template.md:25]: I think we can remove this item

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+- [ ] Version has been bumped. Release notes have been updated. Is this a breaking change?
```
Comment (eamahanna) [inline .github/pull_request_template.md:26]: I don't think this is typically going to be applicable.

File: .github/pull_request_template.md (hunk 5 above)
Comment (bio-boris) [reply .github/pull_request_template.md:25]: I don't know which item you are talking about, but I think you mean the jira one. The github UI doesn't make it clear. You need the ticket number in the title of the PR for the JIRA automation to work.

File: .github/pull_request_template.md (hunk 6 above)
Comment (bio-boris) [reply .github/pull_request_template.md:26]: Which one?

Comment (bio-boris) [reply .github/pull_request_template.md]: Maybe about the version? Added that to develop->master pr instructions

File: .github/pull_request_template.md (hunk 4 above)
Comment (MrCreosote) [reply .github/pull_request_template.md]: I guess I have a broader question that follows on from this question: Is the PR meant to be self contained, or is it expected the reader has access to (possibly private) external resources?

I'm imagining a developer in the future looking at some PR trying to diagnose a bug, and either we've stopped using JIRA and our account is gone or the developer can't access the ticket for some other reason (e.g. maybe a 3rd party dev).

Comment (bio-boris) [reply .github/pull_request_template.md]: @MrCreosote It is possible to have multiple types of contributing docs for each repo. I think this is for internal developers only right now. 
Yea it's not open source friendly to link to private tickets in Jira, but I don't think it's a priority, I think this is mostly for internal devs to be able to reference back to the board quickly

**Pull request merged**